### PR TITLE
Login to OM

### PR DESCRIPTION
**Pull request by cdoron**

### CODE REVIEW

**Comment by cdoron** [comment]
1. Made the minimal necessary changes to the OpenMetadata `swagger.json`, to support login. In the future, we should upgrade to the latest `swagger.json`.
1. The code in this PR assumes that the connector must login to OM using username/password. If we would like to supported earlier versions of OM, which support `no-auth`, minor changes to the code would be required.

File: client-swagger/swagger.json [hunk 1]
```diff
@@ -11787,6 +11787,38 @@
...
+          "200" : {
+            "description" : "User Token",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                "$ref" : "#/components/schemas/LoginToken"
+                }
+              }
+            }
+          },
```
Comment (roytman) [inline client-swagger/swagger.json:11815]: you don't want to add e.g. "401" and/or "403" response codes.

File: pkg/openmetadata-connector-core/connector_helper.go [hunk 2]
```diff
@@ -89,7 +91,11 @@ func (s *OpenMetadataAPIService) addTags(ctx context.Context, c *client.APIClien
 
 func (s *OpenMetadataAPIService) PrepareOpenMetadataForFybrik() bool { //nolint
 	ctx := context.Background()
-	c := s.getOpenMetadataClient()
+	c, err := s.getOpenMetadataClient(ctx)
```
Comment (roytman) [inline pkg/openmetadata-connector-core/connector_helper.go:94]: We should allow running multiple OpenMetadata connectors and restart them.

File: pkg/openmetadata-connector-core/connector_service.go [hunk 3]
```diff
@@ -155,7 +161,12 @@ func (s *OpenMetadataAPIService) DeleteAsset(ctx context.Context, xRequestDataca
 		s.initialized = s.PrepareOpenMetadataForFybrik()
 	}
 
-	c := s.getOpenMetadataClient()
+	c, err := s.getOpenMetadataClient(ctx)
+	if err != nil {
+		s.logger.Error().Err(err).Msg(CannotLoginToOM)
+		return api.Response(http.StatusUnauthorized, nil), errors.New(CannotLoginToOM)
```
Comment (roytman) [inline pkg/openmetadata-connector-core/connector_service.go:167]: I would suggest to separate several errors, e.g. server error, wrong URL, wrong request + unauthorized.

File: client-swagger/swagger.json [hunk 4]
```diff
@@ -11787,6 +11787,38 @@
         }
       }
     },
+    "/v1/users/login" : {
+      "post" : {
+        "tags" : [ "users" ],
+        "summary" : "Login User by Password",
+        "description" : "Login a user with Password",
```
[the remark this answers is not in the document]
Comment (cdoron) [reply client-swagger/swagger.json:11794]: I think that this section of swagger.json is automatically generated from:
https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java#L1067

I would rather not change it

File: client-swagger/swagger.json (hunk 1 above)
Comment (cdoron) [reply client-swagger/swagger.json:11815]: I would rather not touch this section of `swagger.json`, which was auto-generated

File: pkg/openmetadata-connector-core/connector_helper.go (hunk 2 above)
Comment (cdoron) [reply pkg/openmetadata-connector-core/connector_helper.go:94]: I don't think that this code prevents multiple OpenMetadata connectors. Let's discuss.

File: pkg/openmetadata-connector-core/connector_service.go (hunk 3 above)
Comment (cdoron) [reply pkg/openmetadata-connector-core/connector_service.go:167]: I don't know how to differentiate between these different problems.
Hopefully, `err` contains useful information.

**Review by roytman — APPROVED** [review]

**Pull request merged**